src/providers/wfs/qgsoapifprovider.cpp

```diff
@@ -0,0 +1,789 @@
+/***************************************************************************
+    qgsoapifprovider.cpp
+    ---------------------
+    begin                : October 2019
+    copyright            : (C) 2019 by Even Rouault
+    email                : even.rouault at spatialys.com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsexpressionnodeimpl.h"
+#include "qgslogger.h"
+#include "qgsmessagelog.h"
+#include "qgsoapifprovider.h"
+#include "qgsoapiflandingpagerequest.h"
+#include "qgsoapifapirequest.h"
+#include "qgsoapifcollection.h"
+#include "qgsoapifitemsrequest.h"
+
+#include <algorithm>
+
+const QString QgsOapifProvider::OAPIF_PROVIDER_KEY = QStringLiteral( "OAPIF" );
+const QString QgsOapifProvider::OAPIF_PROVIDER_DESCRIPTION = QStringLiteral( "OGC API - Features data provider" );
+
+
+QgsOapifProvider::QgsOapifProvider( const QString &uri, const ProviderOptions &options )
+  : QgsVectorDataProvider( uri, options ),
+    mShared( new QgsOapifSharedData( uri ) )
+{
+
+  connect( mShared.get(), &QgsOapifSharedData::raiseError, this, &QgsOapifProvider::pushErrorSlot );
+  connect( mShared.get(), &QgsOapifSharedData::extentUpdated, this, &QgsOapifProvider::fullExtentCalculated );
+
+  if ( uri.isEmpty() )
+  {
+    mValid = false;
+    return;
+  }
+
+  mShared->mSourceCrs = QgsCoordinateReferenceSystem::fromOgcWmsCrs( QStringLiteral( "EPSG:4326" ) );
+
+  mSubsetString = mShared->mURI.filter();
+
+  if ( !init() )
+  {
+    mValid = false;
+    return;
+  }
+
+  QString errorMsg;
+  if ( !mShared->computeServerFilter( errorMsg ) )
+  {
+    QgsMessageLog::logMessage( errorMsg, tr( "OAPIF" ) );
+    mValid = false;
+    return;
+  }
+
+  qRegisterMetaType<QgsRectangle>( "QgsRectangle" );
```

Comment:
Should not be required, or if it is, moved to QgsApplication alongside the other registrations

Reply:
was not needed indeed.